File: src/main/java/com/meti/Attribute.java
package com.meti;

public interface Attribute {
	default Field computeField() throws AttributeException {
		throw new AttributeException("Not a field");
	}

	default boolean computeBoolean() throws AttributeException {
		throw new AttributeException("Not a boolean.");
	}

	default int computeInt() throws AttributeException {
		throw new AttributeException("Not an integer.");
	}

	default String computeString() throws AttributeException {
		throw new AttributeException("Not a string.");
	}

	enum Name {
		Value,
		Sign,
		Bits,
		Content,
		Identity, Type
	}

	enum Type {

	}
}

File: src/main/java/com/meti/Attribute.scala
package com.meti

object Attribute {

	sealed trait Name extends Attribute

	object Name {

		case object Group extends Name

		case object Content extends Name

		case object Value extends Name

	}

}

trait Attribute {
	@throws[AttributeException]
	def computeInput: Input = throw AttributeException("Not input.")

	@throws[AttributeException]
	def computeToken: Token = throw AttributeException("Not a token.")
}